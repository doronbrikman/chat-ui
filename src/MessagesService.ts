const EVENT = 'spotim/chat';

export type Message = {
  avatar: string;
  username: string;
  text: string;
  timestamp?: number;
};

export class MessagesService {
  constructor(private socket: SocketIOClient.Socket) {}

  connect() {
    this.socket.open();
  }

  sendMessage(msg: Message) {
    msg.timestamp = Date.now();
    this.socket.emit(EVENT, msg);
  }

  subscribeToMessages(cb: (msg: Message) => void) {
    this.socket.on(EVENT, cb);
  }

  disconnect() {
    this.socket.close();
  }
}
