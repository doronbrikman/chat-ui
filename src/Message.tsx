import React from 'react';
import { Message, Image } from 'semantic-ui-react';

import { Message as MessageType } from './MessagesService';

type Props = {
  isMe: boolean;
  message: MessageType;
};

export const ChatMessage: React.StatelessComponent<Props> = ({
  isMe,
  message: { username, avatar, text }
}) => (
  <Message icon color={isMe ? 'green' : 'blue'}>
    <Image src={avatar} size="mini" />
    <Message.Content>
      <Message.Header>{username}</Message.Header>
      {text}
    </Message.Content>
  </Message>
);
