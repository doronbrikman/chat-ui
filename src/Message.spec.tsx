import React from 'react';
import { mount } from 'enzyme';

import { ChatMessage } from './Message';
import { Message } from './MessagesService';

test('renders correctly', () => {
  const message: Message = {
    username: 'test',
    avatar: 'image-link',
    text: 'Hello'
  };

  const wrapper = mount(<ChatMessage isMe message={message} />);
  expect(wrapper).toMatchSnapshot();
});
