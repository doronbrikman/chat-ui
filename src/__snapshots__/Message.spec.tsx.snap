// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  isMe={true}
  message={
    Object {
      "avatar": "image-link",
      "text": "Hello",
      "username": "test",
    }
  }
>
  <Message
    color="green"
    icon={true}
  >
    <div
      className="ui green icon message"
    >
      <Image
        as="img"
        size="mini"
        src="image-link"
        ui={true}
      >
        <img
          className="ui mini image"
          src="image-link"
        />
      </Image>
      <MessageContent>
        <div
          className="content"
        >
          <MessageHeader>
            <div
              className="header"
            >
              test
            </div>
          </MessageHeader>
          Hello
        </div>
      </MessageContent>
    </div>
  </Message>
</Component>
`;
